packages/eslint-plugin-nx/src/utils/runtime-lint-utils.ts

```diff
@@ -209,9 +211,23 @@ function isConstraintBanningProject(
   externalProject: ProjectGraphExternalNode,
   constraint: DepConstraint
 ): boolean {
-  return constraint.bannedExternalImports.some((importDefinition) =>
-    parseImportWildcards(importDefinition).test(
-      externalProject.data.packageName
+  const { allowedExternalImports, bannedExternalImports } = constraint;
+  const { packageName } = externalProject.data;
+
+  /* Check if import is banned... */
+  if (
+    bannedExternalImports?.some((importDefinition) =>
+      parseImportWildcards(importDefinition).test(packageName)
+    )
+  ) {
+    return true;
+  }
+
+  /* ... then check if there is a whitelist and if there is a match in the whitelist.  */
+  return (
+    allowedExternalImports != null &&
```

Comment:
This comparison by value got me confused for a bit. Can we replace it with just `allowedExternalImports && ...`?

Perhaps it would be cleaner to use `elvis + every` to be in sync with `bannedExeternalImports`:
```
return allowedExternalImports?.every(importDefinition =>
  !parseImportWildcards(importDefinition).test(packageName)
);
```

Reply:
> This comparison by value got me confused for a bit. Can we replace it with just allowedExternalImports && ...?

I just like to avoid the boolean coercion and make it explicit that I am not **mistakenly** (thinking that `[] == false` just like `'' == false`) trying to check that the array is not empty.
That said, I'm totally alright with the coercing instead and going with `allowedExternalImports && ...`.

Now, concerning optional chaining + `every`, the problem is that it doesn't make it very explicit that we are expecting `allowedExternalImports = []` to return true... which actually happens because `[].every(() => true)` returns true.
That said, the current implementation `!allowedExternalImports.some(...)` doesn't make it more expclit 😅.

Anyway, there is a test that makes it clear we are expecting `allowedExternalImports = []` to make any external package import fail so I am ok with any implementation.
We can switch to `allowedExternalImports?.every(...)` (which ends up being my favorite).